iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ArmBaseConverter.java

```diff
@@ -138,7 +138,10 @@ public NumericLiteral toNumericLiteral(PropertyTree property) {
       throw convertError(property, value, NumericLiteral.class.getSimpleName(), "ScalarTree.Style.PLAIN");
     }
     try {
-      return new NumericLiteralImpl(Float.parseFloat(value.value()), value.metadata());
+      // for validation
+      Double.parseDouble(value.value());
```

Comment:
Is there no smarter validation? Why does the grammar is not able to validate this? Why the factory should try to create a numericLiteral tree which is not valid?

Reply:
Well the way you are supposed to do it (based on documentation) is to copy / paste a giant regex into your code and validate strings with it.
I'm not sure that is better.
This is inside the ArmBaseConverter for Json, so there is no grammar.